### PR TITLE
Add SPM support to sonar-scanner

### PR DESCRIPTION
**Pull request by SarahJmaiel**
Hello This PR allow to support SPM in sonar-scanner step for iOS swift project

### CODE REVIEW

File: step.sh [hunk 1]
```diff
@@ -344,7 +344,7 @@ else
 fi
 
 # SonarQube
-sonarScannerOptions="-Dsonar.host.url=${sonar_host_url} -Dsonar.login=${SONAR_HOST_LOGIN} -Dsonar.projectKey=${project_key} -Dsonar.language=swift -Dsonar.exclusions=${exclusions} -Dsonar.organization=${sonar_host_organization} -Dsonar.projectVersion=${projet_version}"
+sonarScannerOptions="-X -Dsonar.host.url=${sonar_host_url} -Dsonar.login=${SONAR_HOST_LOGIN} -Dsonar.projectKey=${project_key} -Dsonar.language=swift -Dsonar.exclusions=${exclusions} -Dsonar.organization=${sonar_host_organization} -Dsonar.projectVersion=${projet_version}"
```
Comment (jdanthinne) [inline step.sh]: @SarahJmaiel What does the -X option do?

Comment (SarahJmaiel) [reply step.sh]: I've added as configurable option

File: step.sh [hunk 2]
```diff
@@ -344,7 +345,11 @@ else
 fi
 
 # SonarQube
-sonarScannerOptions="-X -Dsonar.host.url=${sonar_host_url} -Dsonar.login=${SONAR_HOST_LOGIN} -Dsonar.projectKey=${project_key} -Dsonar.language=swift -Dsonar.exclusions=${exclusions} -Dsonar.organization=${sonar_host_organization} -Dsonar.projectVersion=${projet_version}"
+if [ "$enableDebug" = "on"]; then
+    sonarScannerOptions="-X "
+fi
+
+sonarScannerOptions="-Dsonar.host.url=${sonar_host_url} -Dsonar.login=${SONAR_HOST_LOGIN} -Dsonar.projectKey=${project_key} -Dsonar.language=swift -Dsonar.exclusions=${exclusions} -Dsonar.organization=${sonar_host_organization} -Dsonar.projectVersion=${projet_version}"
```
Comment (jdanthinne) [inline step.sh]: @SarahJmaiel The "-X " will be overwritten…

Comment (SarahJmaiel) [reply step.sh]: problem fixed

**Pull request merged**